pkg/cloudprovider/provider/hetzner/provider.go

```diff
@@ -183,6 +192,17 @@ func (p *provider) Validate(spec v1alpha1.MachineSpec) error {
 		}
 	}
 
+	if len(c.Firewalls) != 0 {
+		for _, firewall := range c.Firewalls {
+			if f, _, err := client.Firewall.Get(ctx, firewall); err != nil {
```

Comment:
Sorry, my previous suggestion was not fully correct. This should work.

```suggestion
			f, _, err := client.Firewall.Get(ctx, firewall)
			if err != nil {
```